docs/architecture/deep-dives/passkeys/implementations/relying-party/prf.md

```diff
@@ -0,0 +1,101 @@
+specification by which credential assertion can return the output from a pseudo-random function
+(hence "PRF") that is unique to each credential on the authenticator. The Bitwarden relying party
+can leverage this to obtain a unique, cryptographically-secure key, with which it can encrypt and
+decrypt the user's symmetric key - analogous to the derived master key when using a master password
+to log in.
+
+## How does Bitwarden use the PRF key?
+
+### During passkey registration
+
+When a passkey is registered for use with Bitwarden, we interrogate the attestation response to see
+if PRF is supported. If so, we allow the user to choose to enable encryption.
+
+![Registering a passkey with PRF](image.png)
+
+If the user chooses to enable encryption, we request an assertion from the authenticator, which will
+provide the key. (This is why you may be prompted multiple times to provide user verification when
+registering a PRF-capable passkey.)
+
+The 32-byte key returned from the authenticator is then stretched to a 64-byte two-part key using
+HKDF. The first 32 bytes will serve as the AES-256 encryption key and the second 32 bytes will serve
+as a message authentication code (MAC).
+
+Once we have this encryption key, we can encrypt the user's symmetric key as follows:
+
+    1. Create an RSA public/private key pair, specific to this credential.
+    2. Encrypt the RSA private key with the encryption key.
+    3. Encrypt the user's symmetric key with the RSA public key.
+
+The PRF-encrypted private key, the public key-encrypted encryption key, and the RSA public key are
+persisted to the server and associated with the registered passkey. Note that the PRF key does
+**not** leave the client at any point, to ensure zero-knowledge encryption.
+
+:::tip Why an RSA key pair?
+
+Put simply: key rotation. In order to support
+[encryption key rotation](https://bitwarden.com/help/account-encryption-key/#rotate-your-encryption-key),
+a user must be able to re-encrypt the symmetric key for all registered passkeys without gaining
+access to the PRF keys for each registered credential. The use of a public/private keypair allows a
+user to re-encrypt the new symmetric key with the public key, with the assurance that only the
+holder of the private key can decrypt it.
+
+:::
+
+### During passkey login
+
+When a user chooses to authenticate with a passkey, we request an assertion from the authenticator.
+If the passkey is registered for encryption, and the
+[platform supports it](#what-if-i-am-prompted-for-my-master-password-when-using-a-prf-enabled-passkey),
+the Bitwarden client will receive the PRF key in the assertion response.
+
+The client will then submit the assertion (without the PRF key) to the server to perform
+authentication.
+
+If the server is able to validate the assertion, the user is authenticated, and as a part of the
+authentication response the PRF-encrypted key data is returned to the client:
```

Comment:
```suggestion
authentication response the following PRF-encrypted key data is returned to the client:
```

Reply:
I addressed this with [4ab27ff (#307)](https://github.com/bitwarden/contributing-docs/pull/307/commits/4ab27ff50f7a78620ea68d6f99691c481f24c263).